pilot/cmd/pilot-agent/main.go

```diff
@@ -200,7 +200,12 @@ var (
 			proxyConfig.DiscoveryAddress = discoveryAddress
 			proxyConfig.ConnectTimeout = types.DurationProto(connectTimeout)
 			proxyConfig.StatsdUdpAddress = statsdUDPAddress
-			proxyConfig.EnvoyMetricsService = &meshconfig.RemoteService{Address: envoyMetricsServiceAddress}
+			if envoyMetricsService != "" {
+				if ms := fromJSON(envoyMetricsService); ms != nil {
+					proxyConfig.EnvoyMetricsService = ms
```

Comment:
passing json to a flag does not seem like the right approach

Reply:
@howardjohn I took the same approach as Access Log Service does and it seems it was discussed and agreed here https://github.com/istio/istio/pull/16161/files/c1e772217792807d5ecede3140a88ded7ffd3111#r312473860